langlib/lang.function/src/main/java/org/ballerinalang/langlib/function/Call.java

```diff
@@ -64,7 +64,8 @@ public static Object call(BFunctionPointer<Object, Object> func, Object... args)
             throw ErrorCreator.createError(
                         getModulePrefixedReason(FUNCTION_LANG_LIB, INCOMPATIBLE_ARGUMENTS),
                         BLangExceptionHelper.getErrorDetails(RuntimeErrors.INCOMPATIBLE_ARGUMENTS,
-                        new BTupleType(paramTypes, restType, 0, false), new BTupleType(argTypes)));
+                        removeBracketsFromStringFormatOfTuple(new BTupleType(paramTypes, restType, 0, false)),
```

Comment:
Isn't the error message invalid when 
- there are no parameters but there are arguments
- there are no arguments but the function has required parameters

Looks like we don't have enough tests. Please address these and add more tests.

Reply:
Please check on positive tests too.

Reply:
Update the error msg and a few more test cases.